### ISSUE TITLE
Track Dia overwriting

### ISSUE THREAD

**Issue by kkragenbrink** [xivanalysis/xivanalysis#602]
- [ ] If Dia is overwritten while Afflatus Misery is available, it's a big mistake
- [ ] If Dia overwrites while more than X ticks are remaining, and you aren't moving, it should be a suggestion not to do so
- [ ] It's okay to overwrite Dia after 3 ticks if it is used to create a weaving window

**Comment by AzariahAeroborn** [comment]
SMN’s Ruin 2 module probably has some useful logic to steal for this, since it’s generally a “only use for movement or weaving or pet BS” skill, and they have custom checks to recommend based on that.

**Comment by SaltedyLevity** [comment]
Dia is your absolutely last resort to use to move or weave, because you're not just losing ticks but also misaligning with assize and other CDs in your kit, forcing you to most likely clip down the line, so you might want to be a bit more punishing here with the recommendations. But I'm a bit unsure how to approach it.

I think the first line was meaning if you overwrite dia for moving or weaving when you had misery, right? That's true, although it's more like you already shouldn't be using dia for moving/weaving so it's only just like extra bad. Plus depending on the fight it might be held for multiple targets? That part might be too niche to consider though.

Second one is correct, you frankly don't ever want to overwrite it more than like 3s early.

Third one is super subjective/incorrect in the sense that I don't know how you'd write the logic for it. In that, like I said, it's still incorrect to use dia to create weave windows, instead planning ahead/ using afflatus (or regen if regen will save a GCD heal) are the go-to options. It's _less_ bad to overwrite it early near the end of the fight, because you have fewer alignment windows to care about, and thus also more punishing to do so early in the fight because you're async'd for the remainder. Personally I'm not sure this suggestion should be included at all.
Also I think your math here is just to make it more potent than Glare? That's a bit arbitrary because you're still losing 420p potential. There isn't really a "if it did more than Glare it's OK" rule.

tl;dr take away players should get is to very rarely and actively avoid overwriting dia early. I'm a little on the fence about how far to state this one, because while optimized you can actually do this just fine, average players likely won't. So, on the one hand I want to encourage them for better play, but on the other I don't want them to feel like it's impossible?

Let me know what you're thinking!

**Comment by ayyaruq** [comment]
Bit of a necro but, we track DoTs in clips per minute, so getting specific clips or setting specific tick allowances are features that could be added to the core DoTs analyser, prob relevant to #1478. As far as correlating against Misery, if that's even still necessary, the module would need custom handling for that. Popping this here in case any of the more recent WHM contributors are interested in tackling this.
